### PR TITLE
Updated fold

### PR DESCRIPTION
**Pull request by nepalez**
Updated #33 w/o compacting results

### CODE REVIEW

File: lib/transproc/hash.rb [hunk 1]
```diff
@@ -280,5 +280,37 @@ def unwrap!(hash, root, keys = nil)
...
+    #   Transproc(:fold, :tasks, :priority)[source]
+    #   # => { name: "Jane", tasks: [1, nil] }
+    #
+    # @param [Hash]
+    #
+    # @return [Hash]
+    #
+    # @api public
+    def fold(hash, key, tuple_key)
+      hash.merge key => Transproc(:extract_key, tuple_key)[hash[key]]
```
Comment (solnic) [inline lib/transproc/hash.rb]: you should be able to simply call the method here `hash.merge key => extract_key(hash[key], tuple_key)`

Comment (nepalez) [inline lib/transproc/hash.rb]: got it, thanks. will practice new knowledge in a minute

Comment (nepalez) [inline lib/transproc/hash.rb]: no, it is not work here. extract_key is an array transformation, while fold is a hash'es one

Comment (AMHOL) [inline lib/transproc/hash.rb]: You can just `require 'transproc/array'` at the top of the file, if it's dependent on a transformation from another file, should do that

Comment (nepalez) [inline lib/transproc/hash.rb]: have done

**Comment by AMHOL** [comment]
:+1: looks good to me

**Pull request merged**